x-pack/plugin/sql/src/main/java/org/elasticsearch/xpack/sql/querydsl/agg/GroupByScriptKey.java

```diff
@@ -39,12 +40,18 @@ public TermsValuesSourceBuilder asValueSource() {
                 .order(direction().asOrder())
                 .missingBucket(true);
 
-        if (script.outputType().isInteger()) {
+        if (script.outputType() == DataType.BOOLEAN) {
```

Comment:
Why pick this order? Either is alphabetically or usage, in which case, boolean should be at the bottom along with IP and the rest higher up.